server/queues/processors/DocumentSubscriptionProcessor.ts

```diff
@@ -0,0 +1,72 @@
+import { Transaction } from "sequelize";
+import subscriptionCreator from "@server/commands/subscriptionCreator";
+import { createContext } from "@server/context";
+import { Subscription, User } from "@server/models";
+import { sequelize } from "@server/storage/database";
+import { DocumentUserEvent, Event } from "@server/types";
+import BaseProcessor from "./BaseProcessor";
+
+export default class DocumentSubscriptionProcessor extends BaseProcessor {
+  static applicableEvents: Event["name"][] = [
+    "documents.add_user",
+    "documents.remove_user",
+  ];
+
+  async perform(event: DocumentUserEvent) {
+    const user = await User.findByPk(event.userId);
+    if (!user) {
+      return;
+    }
+
+    switch (event.name) {
+      case "documents.add_user": {
+        return this.addUser(event, user);
+      }
+
+      case "documents.remove_user": {
+        return this.removeUser(event, user);
+      }
+
+      default:
+    }
+  }
+
+  private async addUser(event: DocumentUserEvent, user: User) {
+    await sequelize.transaction(async (transaction) => {
+      await subscriptionCreator({
+        ctx: createContext({
```

Comment:
Thinking that we should do a little refactor soon to make a prebuilt `ctx` a parameter of `perform`